libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsStorage.java

```diff
@@ -3,5 +3,187 @@
+                    continue;
+                }
+            } catch (InterruptedException | ExecutionException e) {
+                e.printStackTrace();
+            }
+
+        }
+        return CompletableFuture.supplyAsync(() -> items);
+    }
+
+    public CompletableFuture<Void> write(Map<String, Object> changes) {
+        if (changes == null) {
+            throw new IllegalArgumentException("The 'changes' parameter is required.");
+        }
+
+        for (Map.Entry<String, Object> keyValuePair : changes.entrySet()) {
+            Object newValue = keyValuePair.getValue();
+            StoreItem storeItem = (StoreItem) newValue;
+
+            // "*" eTag in StoreItem converts to null condition for AccessCondition
+            BlobRequestConditions condition = (!StringUtils.isBlank(storeItem.getETag()) && !storeItem.getETag().equals("*"))
+                ? new BlobRequestConditions().setIfMatch(storeItem.getETag())
+                : null;
+
+            String blobName = GetBlobName(keyValuePair.getKey());
+            BlobClient blobReference = containerClient.getBlobClient(blobName);
+
+            try {
+                String json = jacksonAdapter.serialize(newValue);
+                InputStream stream = new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8));
+                //verify the corresponding length
+                blobReference.uploadWithResponse (stream, 41, null, null, null, null,condition, null, Context.NONE);
+            } catch(HttpResponseException ex) {
+                if (ex.getResponse().getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
+                    StringBuilder sb = new StringBuilder("An error occurred while trying to write an object. The underlying ");
+                    sb.append(BlobErrorCode.INVALID_BLOCK_LIST);
+                    sb.append(" error is commonly caused due to concurrently uploading an object larger than 128MB in size.");
+
+                    throw new HttpResponseException(sb.toString(), ex.getResponse());
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        return CompletableFuture.completedFuture(null);
+    }
+
+    private static String GetBlobName(String key)
+    {
+        if (StringUtils.isBlank(key)) {
+            throw new IllegalArgumentException("key");
+        }
+
+        String blobName;
+        try {
+            blobName = URLEncoder.encode(key, StandardCharsets.UTF_8.toString());
+        } catch (UnsupportedEncodingException e) {
+            throw new IllegalArgumentException("The key could not be encoded");
+        }
+
+        return blobName;
+    }
+
+    private CompletableFuture<Object> innerReadBlob(BlobClient blobReference) {
+        Integer i = 0;
+        while (true) {
+            try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+                JacksonAdapter jacksonAdapter = new JacksonAdapter();
```

Comment:
Why are we generating new `jacksonAdapter` if we already have it in the BlobsStorage class?